packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -99,6 +101,42 @@ class AnalyzeCommand extends FlutterCommand {
     argParser.addFlag('fatal-warnings',
         help: 'Treat warning level issues as fatal.',
         defaultsTo: true);
+
+    // Android analysis sections. DO NOT add other arg after this line.
```

Comment:
what happens if someone adds another arg after this?

Reply:
The addAndroidAnalysis adds a separator, so all the other args after this method will be group into one section unless they add a new separator

Reply:
I see. Can you just inline those `addFlag()` invocations here then? That way the presentation is more obvious.

Reply:
will do, If the way this pr expanding the flutter analyze looks good to you, I will go ahead and add test and make this pr ready for review

Reply:
The overall approach LGTM. I think we still need some discussion around the exact flag names and help text, but that shouldn't block writing tests

Reply:
added test